src/Module/Filer/RemoveTag.php

```diff
@@ -21,30 +21,68 @@
 
 namespace Friendica\Module\Filer;
 
+use Friendica\App;
 use Friendica\BaseModule;
+use Friendica\Core\L10n;
+use Friendica\Core\System;
 use Friendica\Database\DBA;
 use Friendica\DI;
 use Friendica\Model\Post;
+use Friendica\Module\Response;
+use Friendica\Navigation\SystemMessages;
 use Friendica\Network\HTTPException;
+use Friendica\Util\Profiler;
 use Friendica\Util\XML;
+use Psr\Log\LoggerInterface;
 
 /**
  * Remove a tag from a file
  */
 class RemoveTag extends BaseModule
 {
+	/** @var SystemMessages */
+	private $systemMessages;
+
+	public function __construct(SystemMessages $systemMessages, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->systemMessages = $systemMessages;
+	}
+
+	protected function post(array $request = [])
+	{
+		System::httpError($this->removeTag($request));
+	}
+
 	protected function content(array $request = []): string
 	{
 		if (!local_user()) {
 			throw new HTTPException\ForbiddenException();
 		}
 
-		$logger = DI::logger();
+		$this->removeTag($request, $type, $term);
+
+		if ($type == Post\Category::FILE) {
+			DI::baseUrl()->redirect('filed?file=' . rawurlencode($term));
+		}
+
+		return '';
+	}
 
+	/**
+	 * @param array       $request The $_REQUEST array
```

Comment:
Plz add a short description, what/why are tags removed

Reply:
Unfortunately I cannot reliably answer this question.